src/core_plugins/kibana/public/context/app.js

```diff
@@ -0,0 +1,100 @@
+import _ from 'lodash';
+
+import uiModules from 'ui/modules';
+import contextAppTemplate from './app.html';
+import './components/loading_button';
+import './components/size_picker/size_picker';
+import {
+  createInitialQueryParametersState,
+  QueryParameterActionsProvider,
+  QUERY_PARAMETER_KEYS,
+} from './query_parameters';
+import { QueryActionsProvider } from './query';
+
+const module = uiModules.get('apps/context', [
+  'elasticsearch',
+  'kibana',
+  'kibana/config',
+  'kibana/notify',
+  'ngRoute',
+]);
+
+module.directive('contextApp', function ContextApp() {
+  return {
+    bindToController: true,
+    controller: ContextAppController,
+    controllerAs: 'contextApp',
+    restrict: 'E',
+    scope: {
+      anchorUid: '=',
+      columns: '=',
+      indexPattern: '=',
+      predecessorCount: '=',
+      successorCount: '=',
+      sort: '=',
+      discoverUrl: '=',
+    },
+    template: contextAppTemplate,
+  };
+});
+
+function ContextAppController($scope, Private) {
+  const queryParameterActions = Private(QueryParameterActionsProvider);
+  const queryActions = Private(QueryActionsProvider);
+
+  this.state = createInitialState(this.discoverUrl);
+
+  this.actions = _.mapValues(Object.assign(
+    {},
+    queryParameterActions,
+    queryActions,
+  ), (action) => (...args) => action(this.state)(...args));
```

Comment:
Can we explicitly define each action here? Iteratively defining each action using `mapValues` makes it more difficult to trace an action from a reference here or in the markup back to its origin. It also seems to be applying a lot of unused methods to `actions` (5 unused methods from `query.js` and all of the methods from `query_parameters.js` seem unused). I think we only need this:

```javascript
  this.actions = {
    setAllRows: queryActions.setAllRows(this.state),
    fetchAllRowsWithNewQueryParameters: queryActions.fetchAllRowsWithNewQueryParameters(this.state),
    fetchMorePredecessorRows: queryActions.fetchMorePredecessorRows(this.state),
    fetchGivenPredecessorRows: queryActions.fetchGivenPredecessorRows(this.state),
    fetchMoreSuccessorRows: queryActions.fetchMoreSuccessorRows(this.state),
    fetchGivenSuccessorRows: queryActions.fetchGivenSuccessorRows(this.state),
  };
```